### PR TITLE
Deprecate plotfile.

### PR DESCRIPTION
**Pull request by anntzer**
as discussed in #12180.

Various replacements are showcased in plotfile_demo_sgskip.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by WeatherGod** [comment]
I see no consensus in that issue to deprecate plotfile(). Based on
thumbs-up, we seem to be even split on deprecating versus reimplementing
using numpy functions.

**Comment by timhoffm** [comment]
Deprecating vs rewriting was 50/50. I was for deprecating because `plotfile()` does too much and is beyond the scope of matplotlib. It's a chance to cut back on our API.

Disregarding my personal preference, something has to be done about `plotfile()`. Unless someone volunteers to rewrite it, this deprecation is the practical way to go.

File: examples/misc/plotfile_demo_sgskip.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+"""
+=============
+plotfile demo
+=============
+
+Replacing the deprecated `plotfile` by pandas or other matplotlib plotting
```
Comment (timhoffm) [inline examples/misc/plotfile_demo_sgskip.py]: @anntzer This is really a bit difficult to follow. At first glance I was just seeing `plotfile` all over the place. Are you happy with me rewriting the example to better explain/structure it?

**Comment by jklymak** [comment]
I'll officially change my "vote" in #12180.  I think `plotfile` should go away.  Parsing a file is well outside of matplotlib's scope, and much better handled by other libraries.  

IF an example is changed, I'd prefer the example uses `numpy.csvread` or whatever, and then does the plotting in native plotting commands.  If we want to show the user how to write a "plotfile" wrapper for a certain format of file, that would be great.

File: examples/misc/plotfile_demo_sgskip.py (hunk 1 above)
Comment (anntzer) [reply examples/misc/plotfile_demo_sgskip.py]: @timhoffm go for it.

**Comment by timhoffm** [comment]
@anntzer can you please rebase on top of master.  I'm not sure if I would break things for you if I rebase and then add a commit to enhance the example.

**Comment by anntzer** [comment]
rebased

**Comment by timhoffm** [comment]
Added some more explanation. Once `plotfile` is removed, this can be converted to a general "plotting data from files" example.

**Comment by dopplershift** [comment]
I'll put in a vote for deprecation:
1. The status quo is untenable--we can't be relying on other API's we've deprecated
2. The idea behind `plotfile` is not a good one, it goes back to trying to do everything
3. A quick, nonexhaustive check for `plt.plotfile` on GitHub either gives me false positives or forks of the `plotfile_demo.py` script

So I'm with @timhoffm. Also, if we deprecate and people scream, we can revisit.

**Comment by efiring** [comment]
@timhoffm Are there critical changes needed for this, or can it be merged as-is?

**Comment by timhoffm** [comment]
From my side, this is good enough. We've added a number of examples how to achieve the same functionality as `plotfile` using `numpy` or `pandas`.

**Pull request merged**